rpc/utils.go

```diff
@@ -28,9 +28,9 @@ func isNetworkError(err error) bool {
 
 	errStr := err.Error()
 
-	// Check for specific network-related error patterns
-	return strings.Contains(errStr, "unexpected response type: <nil>. Expected *dispatcherResponse") ||
-		strings.Contains(errStr, "Cannot obtain response during timeout") ||
-		strings.Contains(errStr, "rpc is either down or was not configured") ||
-		strings.Contains(errStr, "Cannot decode response")
+	// Only check for actual DNS-related errors that might be resolved by DNS change
+	// "no such host" - DNS resolution failed for hostname
+	// "lookup" + "timeout" - DNS lookup timeout
+	return strings.Contains(errStr, "no such host") ||
+		(strings.Contains(errStr, "lookup") && strings.Contains(errStr, "timeout"))
```

Comment:
what if we isolate this logic into its own function, something like `isDNSErr(err error) bool`